language/move-lang/src/parser/syntax.rs

```diff
@@ -534,7 +528,12 @@ fn parse_term<'input>(tokens: &mut Lexer<'input>) -> Result<Exp, Error> {
             }
         }
 
-        Tok::True | Tok::False | Tok::U64Value => Exp_::Value(parse_value(tokens)?),
+        Tok::True | Tok::False => Exp_::Value(parse_value(tokens)?),
+        Tok::NumValue => {
+            let i = u128::from_str(tokens.content()).unwrap();
+            tokens.advance()?;
```

Comment:
I believe this will panic in the case that the supplied integer is larger than `u128::MAX`. Error handling here is probably needed.

Reply:
Good catch